### PR TITLE
mbl-master - fix up RPi3 patches for u-boot 2019.10 upgrade

### PR DESCRIPTION
**Pull request by jjohnson-arm**
Refactor all the patches and re-number them based on order of being applied.
Create new disable config for boot modes.

### CODE REVIEW

**Comment by jjohnson-arm** [comment]
Needs more testing, currently trying a dev build: http://jenkins.mbed-linux.arm.com/blue/organizations/jenkins/jij-master/detail/jij-master/9/pipeline

Also noticed patches are from nobody@nowhere.arm.com due to devtool environment, is this a problem?

**Comment by jjohnson-arm** [comment]
Testing has not gone well.

File: meta-raspberrypi-mbl/recipes-bsp/u-boot/u-boot/0002-add-linux-serial-revision-dt-support.patch [hunk 1]
```diff
@@ -1,23 +1,16 @@
-source: https://patchwork.ozlabs.org/patch/1045308/
-From: Anton Gerasimov <tossel@gmail.com>
-To: u-boot@lists.denx.de,
-	agraf@suse.de
-Subject: [U-Boot] [PATCH v2 1/1] board: raspberrypi: add serial and revision
+From 1a115015988795bfd939660587f73b55fbc9d265 Mon Sep 17 00:00:00 2001
```
Comment (diego-sueiro) [inline meta-raspberrypi-mbl/recipes-bsp/u-boot/u-boot/0002-add-linux-serial-revision-dt-support.patch]: I think we should keep the original patch header.

File: meta-raspberrypi-mbl/recipes-bsp/u-boot/u-boot/0001-rpi3-Enable-boot-kernel-from-fit-image.patch [hunk 2]
```diff
@@ -1,41 +1,34 @@
-From 0142df58e2c12d40df7d46cdec11ae76cd135fed Mon Sep 17 00:00:00 2001
-From: Jun Nie <jun.nie@linaro.org>
-Date: Fri, 27 Jul 2018 09:15:15 +0800
-Subject: [PATCH 2/2] rpi3: Enable boot kernel from fit image
+From 16c39d5769ca95a443e835c2c7a433b255fe3bfd Mon Sep 17 00:00:00 2001
```
Comment (diego-sueiro) [inline meta-raspberrypi-mbl/recipes-bsp/u-boot/u-boot/0001-rpi3-Enable-boot-kernel-from-fit-image.patch]: I think we should keep the original patch header.

**Comment by jjohnson-arm** [comment]
Updated patches to use original headers.

**Comment by jjohnson-arm** [comment]
Build in jenkins works now that RPI3 pipeline is fixed:
* jenkins.mbed-linux.arm.com/blue/organizations/jenkins/jij-master/13
* jenkins.mbed-linux.arm.com/blue/organizations/jenkins/jij-master/14 - minor refactor tested locally

**Pull request merged**